evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,260 @@
 import logging
-from typing import List
+from typing import Dict, List
 
-from fastapi import APIRouter
+from bson import ObjectId
+from evaluate import load
+from fastapi import APIRouter, Body, Depends, FastAPI, HTTPException, Request
+from square_auth.auth import Auth
 
-from evaluator.models import DataSet
+from evaluator import mongo_client
+from evaluator.core.dataset_handler import DatasetDoesNotExistError, DatasetHandler
+from evaluator.models import Dataset, DataSet
+from evaluator.mongo.mongo_client import MongoClient
+from evaluator.routers import client_credentials
+from evaluator.routers.evaluator import get_dataset_metadata
 
 logger = logging.getLogger(__name__)
-
 router = APIRouter(prefix="/dataset")
+auth = Auth()
+
+valid_skill_type = {
+    "multiple-choice": "multiple-choice QA",
+    "extractive-qa": "extractive-qa",
+}
+
+
+@router.post("", status_code=201)
+async def create_dataset(
+    *,
+    dataset: Dataset = Body(
+        examples={
+            "multiple_Choice_qa": {
+                "summary": "Multiple-Choice sample",
+                "description": "Dataset for Multiple-Choice question answering",
+                "value": {
+                    "dataset_name": "commensense_qa",
+                    "skill_type": "multiple-choice",
+                    "metric": "accuracy",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "choices": ["choice.text"],
+                        "answer_index": "1",
+                    },
+                },
+            },
+            "extractive_qa": {
+                "summary": "Extractive sample",
+                "description": "Dataset for Extractive question answering",
+                "value": {
+                    "dataset_name": "quoref",
+                    "skill_type": "extractive-qa",
+                    "metric": "squad",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "context": "context",
+                        "answers": ["answers.text"],
+                    },
+                },
+            },
+        },
+    ),
+):
+    # check if the skill-type is not empty
+    # download the dataset and check if the dataset_name skill-type and metric exist on huggingface
+    logger.debug(f"dataset object: {dataset}")
+    try:
+        dataset_handler = DatasetHandler()
+        dataset_downloaded = dataset_handler.download_dataset(dataset.dataset_name)
+
+    except Exception as exception:
+        raise HTTPException(
+            404,
+            f"The Name of the dataset : {dataset.dataset_name} not found on huggingface!",
+        )
+
+    # Load the metric from HuggingFace
+    try:
```

Comment:
Es gibt eine schon Funktion `metric_exists` in `task_helper.py`

Reply:
dann würde ich die Funktion nutzen

Reply:
Du hast mir früher gesagt, dass ich prüfen sollte, ob die metric in huggingface existiert nicht local?

Reply:
die Funktion load_metric überprüft nicht, ob die metric auf Huggingface existiert, sondern gibt nur die metric zurück. Ich habe an dieser Stelle die load() Funcktion  zur Überprüfung aufgerufen.